### PR TITLE
Ts conversion iteration 2

### PR DESCRIPTION
**Pull request by pdeleastar**

### CODE REVIEW

File: src/components/chart/chartSaga.ts [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
+function* chartWorker(action: SelectedSymbol) {
+  const { payload } = action
   const chart = yield call(api.getChart, payload);
   yield put({ type: FETCH_CHART_SUCCESS, payload: chart });
 }
 
-function* filterChartWorker({payload}) {
-  console.log(payload)
+function* filterChartWorker(action: FilterAction) {
+  const { payload } = action
```
Comment (voliva) [inline src/components/chart/chartSaga.ts]: Both ways are right, but you can also destructure in the parameter:
```
function* filterChartWorker({ payload }: FilterAction) {
```

File: src/components/chart/utils/formatChartData.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+// @ts-ignore
```
Comment (voliva) [inline src/components/chart/utils/formatChartData.ts]: Why do you need this line?

File: src/components/chart/utils/formatChartData.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import date from "date-and-time";
+
+interface Item {
+  minute: string, 
+  date: string, 
+  close: number
+}
+
+export function formatChartData(data: [] = []) {
```
Comment (voliva) [inline src/components/chart/utils/formatChartData.ts]: It's better to specify here the type of the array - else `data` is just  `any[]`

If you do it this way, it gets easier to understand what does this function take as parameters, and then you won't need to specify it inside the map below, so you can just do
```
data.map(item => {
```

(typescript can figure it out then)

File: src/components/search/types.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+export default interface SearchState {
+    symbols: [],
```
Comment (voliva) [inline src/components/search/types.ts]: This one is also missing a specific type of array.

File: src/components/chart/chartSaga.ts (hunk 1 above)
Comment (pdeleastar) [reply src/components/chart/chartSaga.ts]: Nice, that gets rid of the next line, I have added that now

File: src/components/chart/utils/formatChartData.ts (hunk 3 above)
Comment (pdeleastar) [reply src/components/chart/utils/formatChartData.ts]: Added to project now

File: src/components/chart/utils/formatChartData.ts (hunk 2 above)
Comment (pdeleastar) [reply src/components/chart/utils/formatChartData.ts]: The library after (date-and-time) does not have a types

File: src/components/search/types.ts (hunk 4 above)
Comment (pdeleastar) [reply src/components/search/types.ts]: Added to project now

**Pull request merged**